### PR TITLE
Install scripts, build app with rollup & configurable mirror app id

### PR DESCRIPTION
**Pull request by Benjamin-Dobell**
"Installation" at this point is developer centric, and probably not what your average user thinks of as installation i.e. You still very much need a shell accessible installation of Node.js etc.

However, the process does work reasonably well, at least on macOS. I've made an effort to support Linux (different paths) and Windows (registry), but haven't gotten around to testing those platforms just yet.

In terms of making things end-user installable, I have taken some steps in the direction of being able to build redistributables. Namely the app is now being bundled with rollup, and there's a few configuration options (which I probably ought to document).

For example I've added a `--dependencies` flag that can be used when building app bundle. When enabled all the dependencies are bundled into the app.js bundle, so there's no need to redistribute `node_modules`. However, it doesn't quite work _just_ yet as at runtime `castv2` attempts to load protobuf definitions from the file system (from `node_modules`). That issue can likely be addressed with a simple custom rollup plugin that replaces the runtime filesystem loading with regular `require("foo.proto")` statements. Then the `rollup-plugin-protobuf` plugin can parse the protobuf definitions and include them directly in the bundle.

Regarding macOS support (#14), I didn't really have to change much as everything was already cross-platform - basically just needed to tweak `launcher.sh` (to add `/usr/local/bin` into `PATH`) and ensure the manifest is installed to correct location.

This certainly isn't the most polished PR I've ever submitted, mostly just submitting now to get a dialogue going, particularly since I'm fundamentally messing with the way the software is built etc. quite possibly in ways you didn't have in mind.

### CODE REVIEW

**Comment by Benjamin-Dobell** [comment]
Oh, I should note, I made the mirrorcast app ID configurable because I ran into `NOT_FOUND` errors from my Chromecast when using the existing hard-coded app ID. I'm assuming it's an unpublished app. It's now configurable with:

`npm run build --appId <your APP ID>`

In saying that, I didn't actually manage to get tab / screen mirroring working, even with my own app ID. Haven't bothered investigating just yet. However, media casting e.g. Netflix (with a spoofed user agent) works nicely enough.

**Comment by hensm** [comment]
Took a quick look. I’ll test this out later today (currently 3AM). Looks good so far, I’ve got no issues with this direction. 

Not sure about the mirrorcast app, it’s unfinished. You are right, it’s unpublished, so it just works with my Chromecast. The receiver application URL currently points to my local IP for my dev machine, where I have an HTTP server running. It would be preferable to bundle it with the app so I don’t have to deal with versioning, but the URL is hardcoded, so it’s probably impossible. Hosting it on github.io seems like the next best thing.

Been meaning to work some more on this lately, now that I have more free time, so very cool to see some activity.

File: ext/package.json [hunk 1]
```diff
@@ -12,12 +12,14 @@
...
         "babel-preset-react": "^6.24.1",
         "copy-webpack-plugin": "^4.3.1",
+        "minimist": "^1.2.0",
         "react": "^16.2.0",
         "react-dom": "^16.2.0",
         "web-ext": "^2.7.0",
         "webpack": "^3.10.0"
     },
     "dependencies": {
+        "string-replace-loader": "^2.1.1",
```
Comment (Benjamin-Dobell) [inline ext/package.json]: Whoops, was messing around with this to do the mirrorcast APPI ID, but went with Webpack's DefinePlugin plugin (already in used) in the end. Will submit a force push shortly to blast it away.

**Comment by hensm** [comment]
Sorry, didn’t get a chance to try it yesterday, hopefully today.

Not massively important right now, but as far as code style goes, there isn’t a strict guide. It’s pretty much just four-space indents, camel case, comma-first and 80 char column limit. Everything else is more or less discretionary, and there’s some inconsistency.

Could be worth adopting an existing well known style guide to make contributing easier, though?

**Comment by hensm** [comment]
Just tried it out. [Install fails](https://gist.github.com/hensm/cd7aeaf9938878a67db32bb07d33cffb) if the `native-messaging-hosts` directory doesn't exist. Otherwise, no other issues, works as described on Linux. I'll test the Windows install scripts at some point, though it looks okay to me.

I think it's cleaner if we put all the dist files within the top level `dist/`, so maybe dividing that up into `dist/ext/`, `dist/app/`, instead of sub-level `app/dist/` directories, so it's clear what changes during the build process.

`package-lock.json` would be good in a separate commit, though I'm definitely guilty of this if you look at the commit history.

**Comment by hensm** [comment]
True, that is a good point. Something I hadn’t considered. Leave it as is, then.

Personally, I just find lock file diffs make the commit history more difficult to read, since they’re usually massive compared to any other changes. Creates a lot of noise, but there are other ways to filter that out.

**Comment by Benjamin-Dobell** [comment]
I've just force pushed changes which I believe address all your feedback points.

I do apologise if I still managed to stuff up the code style in certain place. It seems to be impossible to setup Jetbrains IDE's to match this project's code style. As such I had to make the changes by hand and was somewhat fighting the IDE along the way. Hopefully it's right!

**Comment by hensm** [comment]
Looks good to me, merged. Thanks!

**Comment by hensm** [comment]
#18 Issue for dependencies flag

**Pull request merged**